iofog-agent-daemon/src/main/java/org/eclipse/iofog/field_agent/FieldAgent.java

```diff
@@ -433,11 +432,157 @@ private void verificationFailed(Exception e) {
+                return edgeResourcesList;
+            }
+
+            String filename = EDGE_RESOURCE_FILE;
+            JsonArray edgeResourcesJson = null;
+            try {
+                if (fromFile) {
+                    edgeResourcesJson = readFile(filesPath + filename);
+                    if (edgeResourcesJson == null) {
+                        return loadEdgeResources(false);
+                    }
+                } else {
+                    JsonObject result = orchestrator.request("edgeResources", RequestType.GET, null, null);
+                    if(result.containsKey("edgeResources")) {
+                        edgeResourcesJson = result.getJsonArray("edgeResources");
+                        saveFile(edgeResourcesJson, filesPath + filename);
+                    } else {
+                        logError("Error loading microservices from IOFog controller",
+                                new AgentUserException("Error loading microservices from IOFog controller"));
+                    }
+                }
+                try {
+                    if (edgeResourcesJson != null){
+                        List<EdgeResource> edgeResources = IntStream.range(0, edgeResourcesJson.size())
+                                .boxed()
+                                .map(edgeResourcesJson::getJsonObject)
+                                .map(containerJsonObjectToEdgeResourcesFunction())
+                                .collect(toList());
+                        edgeResourceManager.setLatestEdgeResources(edgeResources);
+                        edgeResourcesList.addAll(edgeResources);
+                    }
+                } catch (Exception e) {
+                    logError("Unable to parse microservices", new AgentSystemException(e.getMessage(), e));
+                }
+            } catch (CertificateException | SSLHandshakeException e) {
+                verificationFailed(e);
+                logError("Unable to get microservices due to broken certificate",
+                        new AgentSystemException(e.getMessage(), e));
+            } catch (Exception e) {
+                logError("Unable to get microservices", new AgentSystemException(e.getMessage(), e));
+            }
+            logDebug("Finished Loading microservices...");
+            return edgeResourcesList;
+        }
+    }
+
+    /**
+     * maps json object to EdgeResources
+     * @return
+     */
+    private Function<JsonObject, EdgeResource> containerJsonObjectToEdgeResourcesFunction() {
+        return jsonObj -> {
+            EdgeResource edgeResource = new EdgeResource(jsonObj.getInt("id"), jsonObj.getString("name"), jsonObj.getString("version"));
+            Display display = new Display();
+            EdgeInterface edgeInterface = new EdgeInterface();
+            edgeResource.setDescription(jsonObj.get("description").getValueType().equals(JsonValue.ValueType.NULL) ? null : jsonObj.getString("description"));
```

Comment:
I think you could do:
`edgeResource.setDescription(jsonObj.getString("description", null));`